src/scheduler/pbs_sched.cpp

```diff
@@ -570,6 +570,9 @@ connect_svrpool()
 		svr_conns_secondary =  static_cast<svr_conn_t *>(get_conn_svr_instances(clust_secondary_sock));
 
 		for (i = 0; i < num_conf_svrs; i++) {
+			if (svr_conns_primary == NULL || svr_conns_secondary == NULL)
+				break;
+				
 			if (svr_conns_primary[i].state == SVR_CONN_STATE_DOWN || svr_conns_primary[i].state == SVR_CONN_STATE_DOWN) {	
 				if (svr_conns_primary[i].state == SVR_CONN_STATE_DOWN)
```

Comment:
i think there's a bug here, you are checking svr_conns_primary.state twice (line 577 and line 579), one of them should be checking svr_conns_secondary.state right?

Reply:
Yes. Thanks for catching this. Please look into my latest commit for changes.